python/paddle/distributed/auto_parallel/utils.py

```diff
@@ -472,28 +539,408 @@ def load_distributed_checkpoint(checkpoint_path,
             exe.run(startup_program)
             ckpt_path = ['./output/step_10/model_state_rank0.pdmodel', 
                          './output/step_10/model_state_rank1.pdmodel']
-            load_distributed_checkpoint(ckpt_path, main_program)
+            load_checkpoint_into_program(ckpt_path, main_program)
     """
     checkpoint_path = _check_valid_path(checkpoint_path)
     dist_attr_path = _check_valid_path(dist_attr_path)
 
     if checkpoint_path and dist_attr_path:
-        raise NotImplementedError(
-            "Merge&Slice parameter with dist_attr has not been implemented.")
+        all_state_dict_info = load_distributed_state_dict(checkpoint_path)
+        all_pre_dist_attr = load_distributed_attribute(dist_attr_path)
+        all_cur_dist_attr = get_dist_attr(program)
+
+        all_param_dict = all_state_dict_info["model"]
+        addition_info = all_state_dict_info["addition_info"]
+        sliced_param_dict = merge_and_slice_parameter(
+            all_param_dict, all_pre_dist_attr, all_cur_dist_attr)
+        load_parameter_into_program(sliced_param_dict, program)
+
+        if addition_info:
+            return addition_info
 
     elif checkpoint_path:
         assert len(checkpoint_path) == paddle.distributed.get_world_size(), \
-            "The number of checkpoint_path must equal to the number of ranks"
+            "If distributed attribute is not provided, the number of 'checkpoint_path' \
+                must equal to the number of world_size"
+
         rank = paddle.distributed.get_rank()
         state_dict_info = paddle.load(checkpoint_path[rank])
-        state_dict = state_dict_info["model"]
+
+        param_dict = state_dict_info["model"]
+        addition_info = state_dict_info["addition_info"]
+        load_parameter_into_program(param_dict, program)
+
+        if addition_info:
+            return addition_info
+
     else:
         raise ValueError("'checkpoint_path' can not be None.")
 
-    program.set_state_dict(state_dict) if program else \
-        warnings.warn("'Program' is None, parameters will not be loaded.")
 
-    if "addition_info" not in state_dict_info:
-        return
+def load_parameter_into_program(param_dict, program):
```

Comment:
Would it be better to check param_dict whether None? and if param_dict is None, this api will raise an error or init the param value?

Reply:
Done.